pkg/addon/config.go

```diff
@@ -0,0 +1,214 @@
+package addon
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"strconv"
+
+	"github.com/ViaQ/logerr/log"
+	"github.com/openshift/cluster-logging-operator/pkg/apis"
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	elasticsearch "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const addonSecretName = "addon-cluster-logging-operator-parameters"
+
+var ErrNoAddonConfig = errors.New("No addon configiration found")
+
+type AddonConfiguration struct {
+	EsMaxRetention string `data-field:"es-max-retention" default:"7d"`
+	EsMemory       string `data-field:"es-memory" default:"2Gi"`
+	EsNodeCount    string `data-field:"es-node-count" default:"3"`
+	EsStorageClass string `data-field:"es-storage-class" default:"gp2"`
+	EsStorageSize  string `data-field:"es-storage-size" default:"200G"`
+}
+
+func ProcessAddonConfiguration(namespace string) {
+	// Verify if CLO's CR already present in the target namespace
+	cl, err := newClient()
+	if err != nil {
+		log.Error(err, "failed to create client")
```

Comment:
if method fails, not returning error to `main()` ?

Reply:
The original idea that addon-config won't be processed in this case. It shouldn't prevent CLO from starting